test/upgradeableProxy/DispatcherRC4.upgrade.t.sol

```diff
@@ -152,21 +196,35 @@ contract DispatcherRC4MidwayUpgradeTest is ChannelHandShakeUpgradeUtil, UpgradeT
         dummyLightClient2 = new DummyLightClient();
 
         mars = new Mars(dispatcherProxy);
+        uch = deployUCHV2ProxyAndImpl(address(dispatcherProxy));
+        earth = new Earth(address(uch));
+        // Register mw stack and test it isn't corrupted after the upgrade
+        mw1 = new GeneralMiddleware(1 << 1, address(uch));
+        mwBitmap = uch.MW_ID() | mw1.MW_ID();
```

Comment:
I think UCH is actually a dapp itself instead of middleware. At least this is what I recall from the conversation at the last offsite.

Reply:
This test is to ensure that the `mwStackAddrs` stored in the uch aren't impacted by upgrades (to test for storage collisions). Are you suggesting that we change our UCH to get rid of the `mwStackAddrs` mapping (and all other mw related methods in the uch) altogether to get rid of UCH's middleware functionality? I'm not sure how much of what we discussed on the onsite is considered finalized design?